entity-service-impl/src/main/java/org/hypertrace/entity/data/service/EntityDataServiceImpl.java

```diff
@@ -148,19 +165,26 @@ public void getAndUpsertEntities(Entities request, StreamObserver<Entity> respon
     try {
 
       Map<Key, Document> documentMap = new HashMap<>();
+      List<Entity> updatedEntities = new ArrayList<>();
       for (Entity entity : request.getEntityList()) {
         Entity normalizedEntity = this.entityNormalizer.normalize(tenantId, entity);
+        updatedEntities.add(normalizedEntity);
```

Comment:
Aren't these the entities we're about to upsert? So they'd be the new state rather than the old (would also include entities that don't yet exist and are about to be created).